pages/galleri/index.tsx

```diff
@@ -0,0 +1,228 @@
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+  height: 100%;
+  @media only screen and (max-width: 767px) {
+    display: none;
+  }
+`;
+
+const Single = styled('div')`
+  width: 350px;
+  cursor: pointer;
+`;
+
+const SingleImg = styled(Image)`
+  max-width: 100%;
+  transform: scale(1);
+  transition: transform 0.2s ease-in-out;
+
+  width: 100%;
+  height: 100%;
+  object-fit: cover;
+
+  &:hover {
+    position: relative;
+    transform: scale(1.11);
+    z-index: 99;
+    transition: transform 0.2s ease-in-out;
+  }
+`;
+
+const FullScreenImageDiv = styled('div')`
+  width: 850px;
+  cursor: pointer;
+
+  width: cals(100%-40px);
+  height: cals(100%-40px);
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`;
+
+const FullScreenImage = styled(Image)`
+  width: 100%;
+  height: 100%;
+  object-fit: cover;
+
+  max-width: 100%;
+  max-height: 100%;
+  pointer-events: none;
+  -webkit-user-select: none;
+  -ms-user-select: none;
+  user-select: none;
+`;
+
+const Title = styled('h1')`
+  font-weight: bold;
+  font-smoothing: antialiased;
+  font-size: 3rem;
+  margin-bottom: 1rem;
+`;
+
+export default function Test() {
+  const [slideNumber, setSlideNumber] = useState(0);
+  const [openModal, setOpenModal] = useState(false);
+
+  const handleOpenModal = (index: number) => {
+    setSlideNumber(index);
+    setOpenModal(true);
+  };
+
+  // Close Modal
+  const handleCloseModal = () => {
+    setOpenModal(false);
+  };
+
+  // Previous Image
+  const prevSlide = () => {
+    slideNumber === 0
+      ? setSlideNumber(images.length - 1)
+      : setSlideNumber(slideNumber - 1);
+  };
+
+  // Next Image
+  const nextSlide = () => {
+    slideNumber + 1 === images.length
+      ? setSlideNumber(0)
+      : setSlideNumber(slideNumber + 1);
+  };
+
+  const StyledButton = ({
+    name,
+    top = '50%',
+    right = '40px',
+    left,
+    onClick,
+  }: {
+    name: string;
+    top?: string;
+    right?: string;
+    left?: string;
+    onClick: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  }) => {
+    return (
+      <>
+        {/* eslint-disable-next-line*/}
+        {/* @ts-ignore*/}
+        <ion-icon
+          onClick={onClick}
+          style={{
+            position: 'fixed',
+            fontSize: '65px',
+            color: itdageneBlue,
+            top: top,
+            right: right,
+            left: left,
+          }}
+          name={name}
+        />
+      </>
+    );
+  };
+
+  const changeImageOnKey = useCallback(
+    (event) => {
+      if (event.key === 'Escape') {
+        handleCloseModal();
+      }
+      if (event.key === 'ArrowRight') {
+        nextSlide();
+      }
+      if (event.key === 'ArrowLeft') {
+        prevSlide();
+      }
+    },
+    [slideNumber]
+  );
+
+  useEffect(() => {
+    document.addEventListener('keydown', changeImageOnKey);
+    return function cleanup() {
+      document.removeEventListener('keydown', changeImageOnKey);
+    };
+  });
```

Comment:
Hmmm. Are you sure this is the best way to do this?

Reply:
Only way I found out.